x509/x509.go

```diff
@@ -1317,6 +1347,20 @@ func (e *NonFatalErrors) HasError() bool {
 	return len(e.Errors) > 0
 }
 
+// IsFatal indicates whether an error is fatal.
+func IsFatal(err error) bool {
+	if err == nil {
```

Comment:
Do you actually need this nil check? Possibly it doesn't pass the casts below?

Reply:
Yeah, I think so, otherwise `IsFatal(nil)` will fall through to the `return true` at the bottom.